crates/codegen/src/lib.rs

```diff
@@ -129,34 +169,38 @@ pub struct Generator {
 
 impl Generator {
     /// Create a new [`Generator`].
-    pub(crate) fn new() -> Self {
-        Self::default()
+    pub fn new(plugin: Plugin) -> Self {
+        Self {
+            plugin,
+            ..Self::default()
+        }
     }
 
     /// Set the plugin.
-    pub(crate) fn plugin(&mut self, plugin: plugin::Plugin) -> &mut Self {
+    pub fn plugin(&mut self, plugin: plugin::Plugin) -> &mut Self {
```

Comment:
It looks like this could be removed if we're taking the plugin as an argument in the `new` function.